app/javascript/campaigner_facing/emails.js

```diff
@@ -0,0 +1,172 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+
+const NoEmailsFound = () => <p>No emails were found</p>;
+
+const Email = props => {
+  const body = () => ({ __html: props.Body });
+
+  return (
+    <div className="panel panel-default">
+      <div className="panel-heading">
+        <h3 className="panel-title">{props.Subject}</h3>
+        <h3 className="panel-title">{props.CreatedAt}</h3>
+        <h3 className="panel-title">
+          <strong>To: </strong>
+          {props.ToEmails.join(', ')}
+        </h3>
+      </div>
+      <div className="panel-body">
+        <div dangerouslySetInnerHTML={body()} />
+      </div>
+    </div>
+  );
+};
+
+class DownloadForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      submitting: false,
+      submitted: false,
+      error: false,
+    };
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    const re = /\S+@\S+\.\S+/;
+    if (!re.test(this.state.email)) return;
+
+    this.setState({ submitting: true });
+
+    const opts = {
+      method: 'POST',
+      headers: {
+        'X-CSRF-Token': document.querySelector('[name="csrf-token"]').content,
```

Comment:
This could break if an element isn't found?

Reply:
Any post request would fail if said element was missing - its presence is critical.

Reply:
Yeah I figured. looks good! I also ran this file by `flow` and that was the only line it complained about (rest looks clean), in case you want to add the `// @flow` header to the file.